imagebuildah/stage_executor.go

```diff
@@ -627,6 +627,23 @@ func (s *StageExecutor) Run(run imagebuilder.Run, config docker.Config) error {
 		WorkingDir:       config.WorkingDir,
 	}
 
+	// Honor `RUN --network=<>`.
+	switch run.Network {
+	case "host":
+		for _, namespace := range options.NamespaceOptions {
+			if namespace.Name == "network" {
+				namespace.Host = true
```

Comment:
The initialization at line 612 is going to have to create a copy instead of just referencing the existing options slice to keep this from bleeding over to the next Run() call.

Reply:
Does this modify the value in the slice, or just the copy in the `namespace` variable?

Reply:
My intention was value in the slice hence i modified the code to make it more clear.